src/client/lib/hooks/useCmpConsent.ts

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react';
+import { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { onConsentChange } from '@guardian/consent-management-platform';
+
+const hasRequiredConsents = (): Promise<boolean> => {
+  const hasConsentedToAll = (state: ConsentState) => {
+    const consentFlags = state.tcfv2?.consents
+      ? Object.values(state.tcfv2.consents)
+      : [];
+    const vendorConsentFlags = state.tcfv2?.vendorConsents
+      ? Object.values(state.tcfv2.vendorConsents)
+      : [];
+    const isEmpty =
+      consentFlags.length === 0 || vendorConsentFlags.length === 0;
+
+    return !isEmpty && [...consentFlags, ...vendorConsentFlags].every(Boolean);
+  };
+
+  return new Promise((resolve) => {
+    onConsentChange((state) => {
+      if (state.tcfv2) {
+        return resolve(hasConsentedToAll(state));
+      }
+
+      if (state.ccpa) {
+        return resolve(state.ccpa.doNotSell === false);
+      }
+
+      if (state.aus) {
+        return resolve(state.aus.personalisedAdvertising);
+      }
+
+      // this shouldn't ever be hit, but this is here as safety
+      return resolve(false);
+    });
+  });
+};
```

Comment:
This logic has been lifted from DCR. I will add a comment and check for any tests we can port over.